modular/metadata/metadata.go

```diff
@@ -0,0 +1,121 @@
+package metadata
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/bnb-chain/greenfield-storage-provider/base/gfspapp"
+	"github.com/bnb-chain/greenfield-storage-provider/core/module"
+	"github.com/bnb-chain/greenfield-storage-provider/core/rcmgr"
+	"github.com/bnb-chain/greenfield-storage-provider/pkg/log"
+	"github.com/bnb-chain/greenfield-storage-provider/service/metadata"
+)
+
+var (
+	MetadataModularName        = strings.ToLower("Metadata")
+	MetadataModularDescription = "Retrieves sp metadata and info."
+)
+
+const (
+	DefaultMetadataStatisticsInterval = 60
+)
+
+var _ module.Modular = &MetadataModular{}
+
+type MetadataModular struct {
+	baseApp *gfspapp.GfSpBaseApp
+	scope   rcmgr.ResourceScope
+
+	// freeQuotaPerBucket defines the free read quota per bucket
+	freeQuotaPerBucket uint64
+	// maxMetadataRequest defines the max handling metadata request number
+	maxMetadataRequest int64
+	// retrievingRequest defines the handling retrieve request number
+	retrievingRequest int64
+	config            *metadata.MetadataConfig
```

Comment:
all config field come from gfspconfig, because it may be expand, if use meataconfig maybe not convenient

Reply:
fixed